landlab/components/landslides/landslide_probability.py

```diff
@@ -0,0 +1,535 @@
+        HSD fractions values} (none)
+
+    Examples
+    ----------
+    >>> from landlab import RasterModelGrid
+    >>> from landlab.components.landslides import LandslideProbability
+    >>> import numpy as np
+    
+    Create a grid on which to calculate landslide probability.
+    
+    >>> grid = RasterModelGrid((5, 4), spacing=(0.2, 0.2))
+    
+    Check the number of core nodes.
+    
+    >>> grid.number_of_core_nodes
+    6
+    
+    The grid will need some input data. To check the names of the fields
+    that provide the input to this component, use the *input_var_names*
+    class property.
+    
+    >>> sorted(LandslideProbability.input_var_names)  # doctest: +NORMALIZE_WHITESPACE
+    ['soil__density',
+     'soil__internal_friction_angle',
+     'soil__maximum_total_cohesion',
+     'soil__minimum_total_cohesion',
+     'soil__mode_total_cohesion',
+     'soil__thickness',
+     'soil__transmissivity',
+     'topographic__slope',
+     'topographic__specific_contributing_area']
+    
+    Check the units for the fields.
+    
+    >>> LandslideProbability.var_units('topographic__specific_contributing_area')
+    'm'
+    
+    Create an input field.
+    
+    >>> grid.at_node['topographic__slope'] = np.random.rand(grid.number_of_nodes)
+    
+    If you are not sure about one of the input or output variables, you can
+    get help for specific variables.
+    
+    >>> LandslideProbability.var_help('soil__transmissivity')  # doctest: +NORMALIZE_WHITESPACE
+    name: soil__transmissivity
+    description:
+      mode rate of water transmitted through a unit width of
+    saturated soil
+    units: m2/day
+    at: node
+    intent: in
+    
+    Additional required fields for component.
+    
+    >>> scatter_dat = np.random.randint(1, 10, grid.number_of_nodes)
+    >>> grid.at_node['topographic__specific_contributing_area'] = np.sort(
+    ...      np.random.randint(30, 900, grid.number_of_nodes))
+    >>> grid.at_node['soil__transmissivity'] = np.sort(
+    ...      np.random.randint(5, 20, grid.number_of_nodes), -1)
+    >>> grid.at_node['soil__mode_total_cohesion'] = np.sort(
+    ...      np.random.randint(30, 900, grid.number_of_nodes))
+    >>> grid.at_node['soil__minimum_total_cohesion'] = (
+    ...      grid.at_node['soil__mode_total_cohesion'] - scatter_dat)
+    >>> grid.at_node['soil__maximum_total_cohesion'] = (
+    ...      grid.at_node['soil__mode_total_cohesion'] + scatter_dat)
+    >>> grid.at_node['soil__internal_friction_angle'] = np.sort(
+    ...      np.random.randint(26, 40, grid.number_of_nodes))
+    >>> grid.at_node['soil__thickness'] = np.sort(
+    ...      np.random.randint(1, 10, grid.number_of_nodes))
+    >>> grid.at_node['soil__density'] = (2000. * np.ones(grid.number_of_nodes))
+    
+    Instantiate the 'LandslideProbability' component to work on this grid,
+    and run it.
+    
+    >>> LS_prob = LandslideProbability(grid)
```

Comment:
So, we have removed all capital letters used as parameters in doc tests that has an interface with the user. We have made sure that all parameters/fields have lower case letters and followed our coding conventions. However, we are reserving our rights to use capital letters to name the variables that may not be in direct interface with the user. Please let us know if this is an issue because it would be a huge time sink and challenging (to come up with names that define these variables logically but concisely).